=== modified file 'src/application-service-appstore.c'
--- old/src/application-service-appstore.c	2011-04-07 18:33:33 +0000
+++ new/src/application-service-appstore.c	2011-04-08 20:54:15 +0000
@@ -498,7 +498,8 @@
 	if (menu == NULL || id == NULL || category == NULL || status == NULL ||
 	    icon_name == NULL) {
 		g_warning("Notification Item on object %s of %s doesn't have enough properties.", app->dbus_object, app->dbus_name);
-		g_free(app); // Need to do more than this, but it gives the idea of the flow we're going for.
+		if (!app->validated)
+			application_free(app);
 	}
 	else {
 		app->validated = TRUE;

